sky/serve/serve_state.py

```diff
@@ -382,3 +411,83 @@ def total_number_provisioning_replicas() -> int:
         if replica_info.status == ReplicaStatus.PROVISIONING:
             provisioning_count += 1
     return provisioning_count
+
+
+# === Version functions ===
+def add_version(service_name: str) -> int:
+    """Adds a version to the database."""
+
+    _DB.cursor.execute(
+        """\
+        SELECT MAX(version) FROM versions
+        WHERE service_name=(?)""", (service_name,))
+
+    # add_version is only called during update.
+    new_version = _DB.cursor.fetchone()[0] + 1
+
+    _DB.cursor.execute(
+        """\
+        INSERT INTO versions
+        (version, service_name, spec)
+        VALUES (?, ?, ?)""", (new_version, service_name, pickle.dumps(None)))
+    _DB.conn.commit()
+
+    return new_version
```

Comment:
This is still not thread safe, as if two processes are calling this function concurrently, they can still get teh same version number. Can we use the `AUTOINCREMENT` functionality as what we did for `job_id`?

https://github.com/skypilot-org/skypilot/blob/f78a16e254faeb3667916d22d55b073339152d09/sky/skylet/job_lib.py#L274-L288

Reply:
Updated. 
```
    _DB.cursor.execute(
        """\
        INSERT INTO version_specs
        (version, service_name, spec)
        VALUES (
            (SELECT COALESCE(MAX(version), 0) + 1 FROM
            version_specs WHERE service_name = ?), ?, ?)
        RETURNING version""", (service_name, service_name, pickle.dumps(None)))
```